New pull request template (#230)
carpmosia/carpmosia: .github/PULL_REQUEST_TEMPLATE.md

<!-- Upstream guidelines reference: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- Please do not remove any of these sections, read the comments and fill this out like a form -->

> ## Player information

Reviewer: Is this really necessary? To me it feel like you're asking for info about the player posting the pr, rather than the changes themselves.

Author: Its more about player facing information, as to about the pr was too vague and can very much include technical details

<!-- Describe your PR as you would to a regular player, omit technical details.
       Make sure to cover all player facing changes as well and as detailed

>        as you can, as this will be forwarded to Discord later. -->

Reviewer: why the newline here? I don't really have a problem with separating the individual sentences to their own lines, but doing so in the middle of one is jarring to me

Author: Newlining is better than wrapping, this is for readability in the relatively small PR submission window

<!-- Describe your PR as you would to a regular player, omit technical details.
       Make sure to cover all player facing changes as well and as detailed
       as you can, as this will be forwarded to Discord later. -->

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
<!-- Discuss how this would affect game balance or explain why it was changed.
       Link any relevant discussions or issues. -->

## Technical details
<!-- Summary of code changes for easier review. -->
<!-- Describe what code changes you did or will do in this PR and optionally why.
       Feel free to use check boxes to track your progress, for example:
       - [ ] Resprited X and Y to work with this change.
       - [ ] Implemented Z and X to handle Y edge case.
       - [ ] Added a new component for all X to prevent Z. -->

## Media
<!-- Attach media if the PR makes in-game changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

> ## Requirements

Reviewer: Does it mean that with this section removed I could PR the Bee Movie script and be fine, since following the Guidelines will be optional?

Author: No, the guidelines ar still mentioned at the very top

I just find it frustrating as everyone is we had PR at the moment either:
- checks boxes without reading 
- doesn't check boxes at all
- fucks up checking boxes 
- deleting the entire fucking template

Also GitHub shows PR progress on the main  PRs page, and it results in "2 out of 2 tasks complete" spam, and it would be very nice to  actually use check boxes for something more useful

>        prototype renames, and provide instructions for fixing them. -->

Reviewer: again the random newline

Author: Refer to https://github.com/carpmosia/carpmosia/pull/230#discussion_r2714869286

>        Feel free to remove or add as many add, remove, fix, tweak lines as you need.

Reviewer: You ended the previous line on a comma, but this one starts with a capital, is it a new sentence or not?

Author: Mistype

>        If there are no player facing changes, you can remove the whole thing.

Reviewer: ```suggestion
       If there are no player facing changes, you can remove it altogether.
```
